src/components/App/DisplayPuzzleInstances/index.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import styles from './DisplayPuzzleInstances.module.scss';
+import { PuzzleCustom } from '../../../types/api/puzzles/puzzle';
+import { getAllPuzzleInstances } from '../../../services';
+import { PuzzleInstance } from '@prisma/client';
+import { Button } from '../../Global';
+import { QRGenerator } from '../../App';
+import toast from 'react-hot-toast';
+
+type DisplayPuzzleInstancesProps = {
+  puzzlesList: PuzzleCustom[];
+};
+
+const DisplayPuzzleInstances = ({
+  puzzlesList
+}: DisplayPuzzleInstancesProps) => {
+  const [puzzleId, setPuzzleId] = useState<string>('');
+  const [allPuzzleInstances, setAllPuzzleInstances] = useState<PuzzleInstance[]>([]);
+  const [puzzleInstances, setPuzzleInstances] = useState<
+    PuzzleInstance[]
+  >([]);
+
+  const getPuzzleInstances = async () => {
+    setAllPuzzleInstances(await getAllPuzzleInstances());
+  };
+
+  useEffect(() => {
+    getPuzzleInstances();
+  }, [puzzlesList]);
+
+  useEffect(() => {
+    if (!allPuzzleInstances || !setPuzzleInstances) return;
+
+    if (!puzzleId) return setPuzzleInstances(allPuzzleInstances);
+
+    setPuzzleInstances(
+      allPuzzleInstances.filter(instance => instance.puzzleId.toString() === puzzleId)
+    );
+  }, [puzzleId, allPuzzleInstances, setPuzzleInstances]);
+
+  return (
+    <div className={styles.puzzleInstancesWrapper}>
+      {/* HEADER */}
+      <div className={styles.puzzleInstancesHeader}>
+        <div className={styles.puzzleInstancesHeaderTitle}>
+          <h3> See Locations </h3>
+        </div>
+        {/* Select puzzle */}
+        <div className={styles.puzzleInstancesHeaderSelectWrapper}>
+          {puzzlesList.length > 0 ? (
```

Comment:
do we need to check length here? Can't we just use the falsey empty array thing

Reply:
That's right